ui/jobs/jobs_options.ts

```diff
@@ -25,6 +25,26 @@ export interface JobsNonConfigOptions {
   NotifyExpiredProcsInCombat: number;
   NotifyExpiredProcsInCombatSound: 'disabled' | 'expired' | 'threshold';
   CompactView: boolean;
+  OptionAST: 'disabled' | 'onlyBar' | 'enabled';
```

Comment:
I worry that this might be limiting for the future, if we ever want to have more toggles that won't fit in a single type.

I think you could do something like this:

```typescript
// in this data structure
PerJobOption: { [name in Job]?: JobOption };

// elsewhere
export type JobOption = {
  enabled: boolean;
  hideBoxes: boolean;
};
```

In `jobs_config.ts`, you *should* be able to use `default` to turn `options.PerJobOption` into the correct value when reading and use `setterFunc` to turn the ui value back into the `options.PerJobOption` when writing. See `TimeToShowDeathReportSeconds` for a `setterFunc` example, and e8s's `uptimeKnockbackStrat` for a `default` function example.

Let me know if that doesn't make sense. But, I think having this would allow us to have per job "hide box 1 / box 2 / box 3" or hide bars and not boxes or show mp / hp differently etc much more easily.

Reply:
Agreed to this thought but I think it would be better to apply @Echoring 's idea here, dividing the monitoring area into `resouces` (hp, mp, job-specific meters like oath for PLD) and `proc` (including DoT), then we have a structure like:

```
interface JobOption {
  resources: boolean;
  procs: boolean;
}
```

Am I understand it correctly, @Echoring ?

Reply:
I feel like this PR is a step in that direction. My comments above are mostly to make sure that it's possible to follow up and add those sorts of additional config tweaks. Do you agree with that, or do you think this would prevent that?

Mostly just trying to suggest that we don't make Souma-Sumire do all the work in this PR, and that we land something that will enable those possibilities in the future.

Reply:
That's fair, I think what you wrote is good to me. Then me and @Echoring would talk about the improvement in the next PR.

Reply:
Sorry if I added too much friction to this PR with my comments, I just wanted to make sure that this wouldn't create issues for other things that people wanted to do. Hopefully what I asked for makes sense; let me know if it's not possible or if I can help.